File: 1486-xor-operation-in-an-array.js
/**
 * @param {number} n
 * @param {number} start
 * @return {number}
 */
const xorOperation = function(n, start) {
    let xor = 0;

    for (let i = 0; i < n; i++) {
        xor ^= start + 2 * i;
    }

    return xor;
};

console.log(xorOperation(5, 0)) // 8
console.log(xorOperation(4, 3)) // 8

File: 1486-xor-operation-in-an-array.py
# ❓ DESCRIPTION:
# You are given an integer n and an integer start.
# Define an array nums where nums[i] = start + 2 * i (0-indexed) and n == nums.length.
# Return the bitwise XOR of all elements of nums.

# ✅ SOLUTION (100% runtime):
class Solution(object):
    def xorOperation(self, n, start):
        xor = 0
        for i in range(n):
            xor ^= start + 2 * i
        return xor

# 📌 TESTCASE: 
s = Solution()
print(s.xorOperation(5, 0)) # 8
print(s.xorOperation(4, 3)) # 8